idp_task1.cpp

```diff
@@ -1,55 +1,61 @@
 #include "socket_wrapper.hpp"
-#include "util.h"
+#include "socket_exception.hpp"
+#include <iostream>
+#include <string.h>
+#include <arpa/inet.h>
 
 static const int LISTENQ = 16;
 static const int DEFAULT_ERROR = -1;
 static const int SERVER_PORT = 8877;
-static const int ECHO_STR_LEN = 6;                                            
+static const char* ECHO_STR_PREFIX = "echo: ";
+static const int ECHO_STR_LEN = strlen(ECHO_STR_PREFIX);
 
 void createTestTCPServer()
 {
     try 
     {
-    std::unique_ptr<SocketWrapper> server_socket = std::make_unique<SocketWrapper>(AF_INET, SOCK_STREAM, 0, SERVER_PORT);
+        Socket_sptr server_socket = std::make_shared<Socket>(AF_INET, SOCK_STREAM, 0);
 
-    server_socket->Bind();
+        server_socket->Bind(INADDR_ANY, htons(SERVER_PORT));
```

Comment:
Maybe lets make `htons` inside `Bind` function? As a user of this class API, I wouldn't want to be worried about this.
Seems like it will make additional `arpa\inet.h` header unneeded here.